diff --git a/gossip/privdata/distributor.go b/gossip/privdata/distributor.go
index a4018986..e8067432 100644
--- a/gossip/privdata/distributor.go
+++ b/gossip/privdata/distributor.go
@@ -22,6 +22,7 @@ import (
 	proto "github.com/hyperledger/fabric/protos/gossip"
 	"github.com/hyperledger/fabric/protos/ledger/rwset"
 	"github.com/pkg/errors"
+	"github.com/spf13/viper"
 )
 
 // gossipAdapter an adapter for API's required from gossip module
-- 
2.14.3 (Apple Git-98)

