From f96050b4bfaa876c6e9ad49d6a1df9251bd32197 Mon Sep 17 00:00:00 2001
From: Ahmed Sajid <ahmed.sajid@securekey.com>
Date: Wed, 4 Sep 2019 17:15:37 -0400
Subject: [PATCH] Doing fix for Thales PKCS11

---
 bccsp/pkcs11/pkcs11.go | 1 -
 1 file changed, 1 deletion(-)

diff --git a/bccsp/pkcs11/pkcs11.go b/bccsp/pkcs11/pkcs11.go
index 159b26da2..7be71391e 100644
--- a/bccsp/pkcs11/pkcs11.go
+++ b/bccsp/pkcs11/pkcs11.go
@@ -270,7 +270,6 @@ func (csp *impl) generateECKey(curve asn1.ObjectIdentifier, ephemeral bool) (ski
 		pkcs11.NewAttribute(pkcs11.CKA_ID, []byte(keylabel)),
 		pkcs11.NewAttribute(pkcs11.CKA_LABEL, keylabel),
 
-		pkcs11.NewAttribute(pkcs11.CKA_EXTRACTABLE, false),
 		pkcs11.NewAttribute(pkcs11.CKA_SENSITIVE, true),
 	}
 
-- 
2.21.0 (Apple Git-122.2)

